### DISCUSSION TITLE
Get all repositories

### DISCUSSION THREAD

**Discussion by vetemi** [community/community#24447] [Apps, API and Webhooks]
<p>I’m trying to get a list of all publicly available projects. I’m new to GraphQL but I was not able to figure out how to achieve this.</p>
<p>I need this for my master thesis, in which I try to apply some Machine Learning stuff on Code of Github. I saw papers in which they did similar things and I’m wondering how they achieved to receive Code of all projects.</p>
<p>Is this even possible? If yes, are they any resources for this?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>If you’re trying to analyze the code of all public repositories on GitHub, probably the best way to do that is via the <a href="https://cloud.google.com/blog/products/gcp/github-on-bigquery-analyze-all-the-open-source-code" rel="nofollow noopener">GitHub dataset on BigQuery</a>. It was specifically designed with researchers in mind.</p>
<p>I hope that helps! Let us know if you have more questions.</p>
